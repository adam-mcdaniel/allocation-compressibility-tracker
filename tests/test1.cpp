#include <stdio.h>
#include <stdlib.h>
#include <chrono>
#include <thread>

int main() {
    for (int i=0; i<250; i++) {
        void *ptr = malloc(10);
        std::this_thread::sleep_for(std::chrono::milliseconds(50));
    }
    printf("Hello World!\n");
    return 0;
}