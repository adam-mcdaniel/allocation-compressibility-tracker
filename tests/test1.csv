Iteration (10000 millisecond intervals), Bucket Size, Total Compressed Size, Total Uncompressed Size, Number of Compressions, Compression Ratio
1,10-16,2211,6432,201,0.34375
1,16-32,86,7008,3,0.0122717
1,32-64,89,10976,2,0.0081086
1,64-128,103,7104,1,0.0144989
2,10-16,2211,6432,201,0.34375
2,16-32,86,7008,3,0.0122717
2,32-64,89,10976,2,0.0081086
2,64-128,103,7104,1,0.0144989
