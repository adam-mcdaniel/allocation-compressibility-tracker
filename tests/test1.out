Checking compression stats
Protecting allocations
[FAULT] Working thread, giving back access: 0x7efe8dc80000
Done protecting allocations
[FAULT] Working thread, giving back access: 0x7efe91880228
Unprotecting allocations
Done unprotecting allocations
Updated buckets
Done
Bucket stats:
Bucket 0 (1 bytes)
  No entries
Bucket 1 (2 bytes)
  No entries
Bucket 2 (3 bytes)
  No entries
Bucket 3 (4 bytes)
  No entries
Bucket 4 (5 bytes)
  No entries
Bucket 5 (6 bytes)
  No entries
Bucket 6 (7 bytes)
  No entries
Bucket 7 (8 bytes)
  No entries
Bucket 8 (9 bytes)
  No entries
Bucket 9 (10 bytes)
  No entries
Bucket 10 (16 bytes)
  Total compressed size: 2211
  Total uncompressed size: 6432
  Number of entries: 201
  Compression ratio (lower is better): 0.34375
Bucket 11 (32 bytes)
  Total compressed size: 86
  Total uncompressed size: 7008
  Number of entries: 3
  Compression ratio (lower is better): 0.0122717
Bucket 12 (64 bytes)
  Total compressed size: 89
  Total uncompressed size: 10976
  Number of entries: 2
  Compression ratio (lower is better): 0.0081086
Bucket 13 (128 bytes)
  Total compressed size: 103
  Total uncompressed size: 7104
  Number of entries: 1
  Compression ratio (lower is better): 0.0144989
Bucket 14 (256 bytes)
  No entries
Bucket 15 (512 bytes)
  No entries
Bucket 16 (1024 bytes)
  No entries
Bucket 17 (2048 bytes)
  No entries
Bucket 18 (4096 bytes)
  No entries
Hello World!
Bucket stats:
Bucket 0 (1 bytes)
  No entries
Bucket 1 (2 bytes)
  No entries
Bucket 2 (3 bytes)
  No entries
Bucket 3 (4 bytes)
  No entries
Bucket 4 (5 bytes)
  No entries
Bucket 5 (6 bytes)
  No entries
Bucket 6 (7 bytes)
  No entries
Bucket 7 (8 bytes)
  No entries
Bucket 8 (9 bytes)
  No entries
Bucket 9 (10 bytes)
  No entries
Bucket 10 (16 bytes)
  Total compressed size: 2211
  Total uncompressed size: 6432
  Number of entries: 201
  Compression ratio (lower is better): 0.34375
Bucket 11 (32 bytes)
  Total compressed size: 86
  Total uncompressed size: 7008
  Number of entries: 3
  Compression ratio (lower is better): 0.0122717
Bucket 12 (64 bytes)
  Total compressed size: 89
  Total uncompressed size: 10976
  Number of entries: 2
  Compression ratio (lower is better): 0.0081086
Bucket 13 (128 bytes)
  Total compressed size: 103
  Total uncompressed size: 7104
  Number of entries: 1
  Compression ratio (lower is better): 0.0144989
Bucket 14 (256 bytes)
  No entries
Bucket 15 (512 bytes)
  No entries
Bucket 16 (1024 bytes)
  No entries
Bucket 17 (2048 bytes)
  No entries
Bucket 18 (4096 bytes)
  No entries
Elapsed time: 2570526 microseconds
